### ISSUE TITLE
False Positive from dict-iter-method outside dict context

### ISSUE THREAD

**Issue by tiagohonorato** [PyCQA/pylint#4174]
### Steps to reproduce

Run the following with --py3k:

```
class Test():
    def iteritems(self):
        pass

test = Test()
test.iteritems()
```

### Current behavior

```
W1620: Calling a dict.iter*() method (dict-iter-method)
```
### Expected behavior

Only issue ``W1620`` from ``dict`` instances.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.7 (default, Jan 20 2021, 00:00:00) 
[GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
```

**Comment by Pierre-Sassoulas** [comment]
Won't do, we removed the python3 porting mode in 2.11.
